templates/intro-workspaces/README.ipynb

```diff
@@ -0,0 +1,334 @@
+
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Introduction to Workspaces\n",
+    "\n",
+    "Welcome! You are currently in a Workspace, which is a persistent cloud IDE connected to a Ray cluster.\n",
+    "\n",
+    "In this tutorial, you will learn:\n",
+    "1. Basic workspace features such as git repo persistence, NFS mounts, cloud storage, and SSH authentication.\n",
+    "2. Ray cluster management features, such as adding multiple worker nodes.\n",
+    "3. Ray monitoring features such as viewing tasks in the dashboard.\n",
+    "4. Dependency management.\n",
+    "\n",
+    "## \"Hello world\" in workspaces\n",
+    "\n",
+    "Let's start by checking that Ray is working properly in your workspace. You can do this by running the following cell to execute a simple parallel Ray program."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ray\n",
+    "\n",
+    "@ray.remote\n",
+    "def square(x):\n",
+    "    return x ** 2\n",
+    "\n",
+    "futures = [square.remote(x) for x in range(100)]\n",
+    "results = ray.get(futures)\n",
+    "print(\"Success!\", results)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Workspace Basics\n",
+    "\n",
+    "A Workspace is a cloud IDE where you can develop and test Ray programs. Let's get started by creating a new git repo in this workspace. Workspaces will persist the tracked files in this git repo across restarts (as well as files not in a git repos).\n",
+    "\n",
+    "We'll use the repo later on to author and run a simple Ray app."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!mkdir my_repo && cd my_repo && git init"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setting up SSH authentication (optional)\n",
+    "\n",
+    "Anyscale generates a unique SSH key per user, which is accessible at `~/.ssh/id_rsa.pub`. If you'd like, you can add this key to GitHub in order to access private repositories from Workspaces.\n",
```

Comment:
Shall we add a link to gh doc about setting up ssh? https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account